x/ibc-transfer/keeper/genesis.go

```diff
@@ -0,0 +1,42 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/ibc-transfer/types"
+)
+
+// InitGenesis initializes the ibc-transfer state and binds to PortID.
+func (k Keeper) InitGenesis(ctx sdk.Context, state types.GenesisState) {
+	k.SetPort(ctx, state.PortID)
+
+	for _, trace := range state.DenomTraces {
+		k.SetDenomTrace(ctx, trace)
+	}
+
+	// Only try to bind to port if it is not already bound, since we may already own
```

Comment:
I realise this change isn't from this PR, but reading this line I wonder why we can't always do this in one place, not two?

Reply:
I see, that seems fine.

Reply:
I think it makes sense to encourage app developers to add their ports to the capability genesis upon a new chain or new addition upon restart. This would reduce redundant code app modules write (since every module would need to insert this code) and would centralize this binding. I think it could even make sense to remove this check and enforce that capability genesis is used.